plugins/core/IO/qLASIO/include/LasScalarFieldLoader.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+//##########################################################################
+//#                                                                        #
+//#                CLOUDCOMPARE PLUGIN: LAS-IO Plugin                      #
+//#                                                                        #
+//#  This program is free software; you can redistribute it and/or modify  #
+//#  it under the terms of the GNU General Public License as published by  #
+//#  the Free Software Foundation; version 2 of the License.               #
+//#                                                                        #
+//#  This program is distributed in the hope that it will be useful,       #
+//#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#                   COPYRIGHT: Thomas Montaigu                           #
+//#                                                                        #
+//##########################################################################
+
+#ifndef LASSCALARFIELDLOADER_H
+#define LASSCALARFIELDLOADER_H
+
+#include "LasDetails.h"
+
+#include <FileIOFilter.h>
+#include <QFileInfo>
+
+#include <ccPointCloud.h>
+
+#include <laszip/laszip_api.h>
+
+/// Class with the logic on how to load LAS dimensions values
+/// from the LAS file into a ccPointCloud's scalar field.
+///
+/// This also handle LAS extra scalar fields, as well as RGB.
+class LasScalarFieldLoader
+{
+  public:
+    LasScalarFieldLoader(std::vector<LasScalarField> standardScalarFields,
+                         std::vector<LasExtraScalarField> extraScalarFields,
+                         ccPointCloud &pointCloud);
+
+    CC_FILE_ERROR handleScalarFields(ccPointCloud &pointCloud, const laszip_point &currentPoint);
+
+    CC_FILE_ERROR handleRGBValue(ccPointCloud &pointCloud, const laszip_point &currentPoint);
+
+    CC_FILE_ERROR handleExtraScalarFields(ccPointCloud &pointCloud, const laszip_point &currentPoint);
+
+    const std::vector<LasScalarField> &standardFields() const
+    {
+        return m_standardFields;
+    }
+
+  private:
+    /// Handles loading of LAS value into the scalar field that will be part
+    /// of the pointCloud.
+    ///
+    /// sfInfo: Info about the current scalar field we are loading the value into
+    /// pointCloud: The point cloud where the scalar field will be loaded into
+    /// currentValue: The current value of the LAS field we are loading.
+    template <typename T>
+    CC_FILE_ERROR handleScalarField(LasScalarField &sfInfo, ccPointCloud &pointCloud, T currentValue);
+
+    /// Same thing as `handleScalarField` but for RGB.
```

Comment:
Isn't this method dealing with GPS data?

Reply:
yup